api/v1alpha1/ext_proc_types.go

```diff
@@ -0,0 +1,102 @@
+	StreamedExtProcHeaderProcessingMode        ExtProcBodyProcessingMode = "STREAMED"
+	BufferedExtProcHeaderProcessingMode        ExtProcBodyProcessingMode = "BUFFERED"
+	BufferedPartialExtProcHeaderProcessingMode ExtProcBodyProcessingMode = "BUFFERED_PARTIAL"
+)
+
+type ProcessingModeOptions struct {
+	// Defines header processing mode
+	Headers *ExtProcHeaderProcessingMode `json:"request,omitempty"`
+	// Defines body processing mode
+	Body *ExtProcBodyProcessingMode `json:"response,omitempty"`
+}
+
+// ExtProcProcessingMode defines if and how headers and bodies are sent to the service.
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/ext_proc/v3/processing_mode.proto#envoy-v3-api-msg-extensions-filters-http-ext-proc-v3-processingmode
+type ExtProcProcessingMode struct {
+	// Defines header and body treatment for requests
+	// Default: SEND
+	Request *ProcessingModeOptions `json:"request,omitempty"`
+	// Defines header and body treatment for responses
+	// Default: NONE
+	Response *ProcessingModeOptions `json:"response,omitempty"`
+}
+
+// ExtProcAttributes defines which attributes are
+type ExtProcAttributes struct {
+	// defines attributes to send for Request processing
+	Request []string `json:"request,omitempty"`
+	// defines attributes to send for Response processing
+	Response []string `json:"response,omitempty"`
+}
+
+// MetadataNamespaces defines metadata namespaces that can be used to forward or receive dynamic metadata
+type MetadataNamespaces struct {
+	// Specifies a list of metadata namespaces whose values, if present, will be passed to the ext_proc service as an opaque protobuf::Struct.
+	Untyped []string `json:"untyped,omitempty"`
+	// Specifies a list of metadata namespaces whose values, if present, will be passed to the ext_proc service as a protobuf::Any.
+	Typed []string `json:"typed,omitempty"`
+}
+
+// ExtProcMetadataOptions defines options related to the sending and receiving of dynamic metadata
+type ExtProcMetadataOptions struct {
+	ForwardingNamespaces []MetadataNamespaces `json:"forwardingNamespaces,omitempty"`
+	ReceivingNamespaces  []MetadataNamespaces `json:"receivingNamespaces,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.group) || self.service.backendRef.group == \"\") : true", message="group is invalid, only the core API group (specified by omitting the group field or setting it to an empty string) is supported"
+// +kubebuilder:validation:XValidation:rule="has(self.service) ? (!has(self.service.backendRef.kind) || self.service.backendRef.kind == 'Service') : true", message="kind is invalid, only Service (specified by omitting the kind field or setting it to 'Service') is supported"
+//
+// ExtProc defines the configuration for External Processing.
+type ExtProc struct {
+	// Service defines the configuration of the external processing service
+	Service *ExtProcService `json:"service"`
```

Comment:
If `Service` is an optional field in the api, add the `omitempty` attribute to the field tags.
If it isn't, don't use a pointer here.